pkg/config/policy.go

```diff
@@ -0,0 +1,249 @@
+	if !ok {
+		return nil, errors.New("application missing from config")
+	}
+
+	return getPolicies(application.Policies)
+}
+
+// GetPoliciesForApplicationOrg returns a map of policies for specific application
+// organization.
+func GetPoliciesForApplicationOrg(config cb.Config, orgName string) (map[string]*Policy, error) {
+	orgGroup, ok := config.ChannelGroup.Groups[ApplicationGroupKey].Groups[orgName]
+	if !ok {
+		return nil, fmt.Errorf("application org %s does not exist in channel config", orgName)
+	}
+
+	return getPolicies(orgGroup.Policies)
+}
+
+// getPolicies returns a map of Policy from given map of ConfigPolicy in organization config group.
+func getPolicies(policies map[string]*cb.ConfigPolicy) (map[string]*Policy, error) {
+	if policies == nil {
+		return map[string]*Policy{}, nil
+	}
+
+	p := map[string]*Policy{}
+	for name, policy := range policies {
+		switch policy.Policy.Type {
+		case int32(cb.Policy_IMPLICIT_META):
+			imp := &cb.ImplicitMetaPolicy{}
+			err := proto.Unmarshal(policy.Policy.Value, imp)
+			if err != nil {
+				return nil, err
+			}
+			rule, err := implicitMetaToString(imp)
+			if err != nil {
+				return nil, err
+			}
+
+			p[name] = &Policy{
+				Type: ImplicitMetaPolicyType,
+				Rule: rule,
+			}
+
+		case int32(cb.Policy_SIGNATURE):
+			sp := &cb.SignaturePolicyEnvelope{}
+			err := proto.Unmarshal(policy.Policy.Value, sp)
+			if err != nil {
+				return nil, err
+			}
+			rule, err := signatureMetaToString(sp)
+			if err != nil {
+				return nil, err
+			}
+
+			p[name] = &Policy{
+				Type: SignaturePolicyType,
+				Rule: rule,
+			}
+
+		default:
+			return nil, fmt.Errorf("unknown policy type: %v", policy.Policy.Type)
+		}
+	}
+	return p, nil
+}
+
+// implicitMetaToString converts a *cb.ImplicitMetaPolicy to a string representation.
+func implicitMetaToString(imp *cb.ImplicitMetaPolicy) (string, error) {
+	args := []string{}
+
+	switch imp.Rule {
+	case cb.ImplicitMetaPolicy_ANY:
+		args = append(args, cb.ImplicitMetaPolicy_ANY.String())
+	case cb.ImplicitMetaPolicy_ALL:
+		args = append(args, cb.ImplicitMetaPolicy_ALL.String())
+	case cb.ImplicitMetaPolicy_MAJORITY:
+		args = append(args, cb.ImplicitMetaPolicy_MAJORITY.String())
+	default:
+		return "", fmt.Errorf("unknown implicit meta policy rule type %v", imp.Rule)
+	}
+
+	args = append(args, imp.SubPolicy)
+
+	return strings.Join(args, " "), nil
+}
+
+// signatureMetaToString converts a *cb.SignaturePolicyEnvelope to a string representation.
+func signatureMetaToString(sig *cb.SignaturePolicyEnvelope) (string, error) {
+	roles := []string{}
+	for _, id := range sig.Identities {
+		role, err := mspPrincipalToString(id)
+		if err != nil {
+			return "", err
+		}
+		roles = append(roles, role)
+	}
+	return signaturePolicyToString(sig.Rule, roles)
+}
+
+// mspPrincipalToString converts a *mb.MSPPrincipal to a string representation.
+func mspPrincipalToString(principal *mb.MSPPrincipal) (string, error) {
+	var res strings.Builder
```

Comment:
nit: do we need a builder? If we do, it’s only used in one block. The declaration should be in that block.

Reply:
builder is used here as I considered this might be a faster way to build strings.
I'm open to any suggestions. 
also moved it into the block with incoming fixes.